IQSS/dataverse-frontend
# Feature/338 form validation library

## What this PR does / why we need it:
Adds form validation library (react-hook-form) to the create dataset form in order to create dynamic fields validation

## Which issue(s) this PR closes:

- Will close https://github.com/IQSS/dataverse-frontend/issues/338 once the sub branch (base branch of this pr) has been merged.

## Special notes for your reviewer:
This PR points to a sub-branch because before targeting the develop branch it will be combined with the issue https://github.com/IQSS/dataverse-frontend/issues/336 which is in this sub-branch and then a final pr with both issues combined will be made to develop. All this to be able to bring up something completely stable to beta environment.

## Suggestions on how to test this:
- Log in, click the create dataset button.
- Submit the form without filling any required field, every required field should show te required error feedback, and the form should focus and scroll to the first input with an error.
- Fill a date field with the wrong format, respective feedback error should be visible.
- Fill an integer field with the wrong format (e.g.: 23.5), respective feedback error should be visible.
- Fill a float field with the wrong format (e.g.: 'hi'), respective feedback error should be visible.
- Fill an email field with a wrong format (e.g.: 'test@'), respective feedback error should be visible.
- Fill an URL field with a worng format (e.g.: 'bad://something'), espective feedback error should be visible.
- Submit the form filling al required fields with the correct format, a new dataset should be created.

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
Yes.
![spa_datasets_create(Desktop 1366)](https://github.com/IQSS/dataverse-frontend/assets/64560524/307892e1-e8ce-406b-900a-1f239f7700b6)

Also I modified at spa level not design system level the styles for the checkbox group, I think if it is successful we could apply it to the design system directly in the future. 
This checkbox group that allows to select multiple values is a temporary replacement of the [select multiple](https://github.com/IQSS/dataverse-frontend/issues/355) not yet done.
Examples below: 👇👇
![Screenshot 2024-04-11 at 14 01 29](https://github.com/IQSS/dataverse-frontend/assets/64560524/1792e9d7-87d3-451b-9b13-81b2b350c412)
![Screenshot 2024-04-11 at 14 01 42](https://github.com/IQSS/dataverse-frontend/assets/64560524/2f63d024-3053-415a-ba93-10123b403732)
![Screenshot 2024-04-11 at 14 01 51](https://github.com/IQSS/dataverse-frontend/assets/64560524/e4ca90a5-ac92-45fb-9175-1626671858fc)

## Is there a release notes update needed for this change?:

## Additional documentation:
I have refactored some form elements of the design system so that they can forward a react ref and pass it to the input itself. This was necessary so that the form validation library could focus and scroll to the first item encountered with a validation error.
